AEPMessaging/Sources/PropositionPayload.swift

```diff
@@ -0,0 +1,44 @@
+/*
+ Copyright 2022 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+import AEPServices
+
+struct PropositionPayload: Codable {
+    var propositionInfo: PropositionInfo
+    var items: [PayloadItem]
+    
+    enum CodingKeys: String, CodingKey {
+        case id
+        case scope
+        case scopeDetails
+        case items
+    }
+    
+    init(from decoder: Decoder) throws {
+        let values = try decoder.container(keyedBy: CodingKeys.self)
+        let id = try values.decode(String.self, forKey: .id)
+        let scope = try values.decode(String.self, forKey: .scope)
+        let scopeDetails = try values.decode([String: AnyCodable].self, forKey: .scopeDetails)
+        
+        propositionInfo = PropositionInfo(id: id, scope: scope, scopeDetails: scopeDetails)
+        items = try values.decode([PayloadItem].self, forKey: .items)
```

Comment:
caution: An issue with array decode is that it's all or none. Would you have a scenario where you need to ignore one or more (invalid) payload items in the array?

Reply:
it's a good callout - to be honest if we end up in a position where only some of the items are invalid, i'd have greater concerns about something being broken in the process.  in that case, i think it's best to just not load any messages. wdyt?